packages/server/lib/project-base.ts

```diff
@@ -160,41 +154,37 @@ export class ProjectBase extends EE {
 
     this._server = new ServerBase(cfg)
 
-    let studioManager: StudioManager | null
+    const studioLifecycleManager = new StudioLifecycleManager()
 
-    if (process.env.CYPRESS_ENABLE_CLOUD_STUDIO || process.env.CYPRESS_LOCAL_STUDIO_PATH) {
-      studioManager = await getAndInitializeStudioManager({
+    if (!cfg.isTextTerminal) {
+      studioLifecycleManager.initializeStudioManager({
         projectId: cfg.projectId,
         cloudDataSource: this.ctx.cloud,
+        cfg,
+        debugData: this.configDebugData,
+        ctx: this.ctx,
       })
 
-      this.ctx.update((data) => {
-        data.studio = studioManager
-      })
+      // Register a listener to update config when studio is ready
+      studioLifecycleManager.onStudioReady((studioManager) => {
+        debug('Studio is ready, updating config with isStudioProtocolEnabled:', studioManager.isProtocolEnabled)
```

Comment:
I don't think we need `isStudioProtocolEnabled` on the config object at all anymore. Which I think means (combined with my other comment) we don't need this callback?

Reply:
Yeah I think we could just check if `this.ctx.coreData.studioLifecycleManager` is populated actually.